bindings/CXX11/adios2/cxx11/Engine.h

```diff
@@ -218,9 +218,7 @@ class Engine
         auto bufferView = static_cast<AdiosView<U>>(data);
 #if defined(ADIOS2_HAVE_KOKKOS) || defined(ADIOS2_HAVE_GPU_SUPPORT)
         auto bufferMem = bufferView.memory_space();
-        auto bufferLayout = bufferView.layout();
         variable.SetMemorySpace(bufferMem);
-        variable.SetArrayLayout(bufferLayout);
```

Comment:
When is `SetArrayLayout()` called in cases when 
`! (mem == MemorySpace::GPU && m_ArrayLayout == ArrayOrdering::Auto)`?

`SetMemorySpace()` only calls it when the user has not set the layout before, like Franz' example you are trying to fix.

Reply:
I will double check and update the PR, at a first glance it doesn't seem like I'm updating the layout for Host pointers. However, the tests pass on both CPU and GPU so I will update this PR in a little bit based on what I find.